script.js

```diff
@@ -293,4 +293,34 @@ function checkForm() {
 form.addEventListener('submit', (e) => {
   e.preventDefault();
   checkForm();
-});
\ No newline at end of file
+});
+
+// Local storage exercise
+function saveData() {
+  const storedInfo = {};
+
+  storedInfo.userName = document.getElementById('name').value;
+  storedInfo.userEmail = document.getElementById('email').value;
+  storedInfo.userMsg = document.getElementById('message').value;
+
+  localStorage.setItem('storedInfo', JSON.stringify(storedInfo));
+
+// let retrievedInfo = JSON.parse(localStorage.getItem('storedInfo'));
+}
+
+function showData() {
+  const showInfo = JSON.parse(localStorage.getItem('storedInfo'));
+
+  const name2 = document.getElementById('name');
+  const email2 = document.getElementById('email');
+  const msg2 = document.getElementById('message');
+
+  name2.value = showInfo.userName;
+  email2.value = showInfo.userEmail;
+  msg2.value = showInfo.userMsg;
+}
```

Comment:
- [ ] Please notice that your website doesn't store the input in localStorage when you refresh the page. this is due to the fact that you don't have an `if` statement that checks whether there is any `localStorage` in the first place. 

for a user who hasn't been to the site before they won't have any `localStorage` so this function will crash when executed. please add an `if` statement that checks whether there is `localStorage` in the first place.

![image](https://user-images.githubusercontent.com/18730106/151454185-04f62450-73c9-4207-a768-1ce8b89ca7cd.png)

![image](https://user-images.githubusercontent.com/18730106/151454206-48e72b49-ff79-4f75-808a-49aecc2660c8.png)